[TASK] Add port mapping and reverse proxy guidance (#5744)
TYPO3-Documentation/TYPO3CMS-Reference-CoreApi: Documentation/Administration/Docker/Production/Ports/Index.rst

:navigation-title: Ports

> ..  include:: /Includes.rst.txt
> .. _container-hosting-ports:

Reviewer: ```suggestion
..  include:: /Includes.rst.txt
..  _container-hosting-ports:
```

===================================================
Ports when running TYPO3 on container-based hosting
===================================================

> 

Author: Good point, added an explanation

> ===================================================
> Ports when running TYPO3 on container-based hosting
> ===================================================
> 
> ..  include:: /Administration/Docker/Production/_Experimental.rst.txt
> 
> When deploying TYPO3 to a container-based hosting platform, make sure that the

Author: ```suggestion
===================================================
Ports when running TYPO3 on container-based hosting
===================================================

..  include:: /Administration/Docker/Production/_Experimental.rst.txt

This section applies to running TYPO3 in managed container environments such as
Kubernetes clusters, Nexaa, AWS App Runner, or Azure App Services. These platforms
often map internal container ports (like `80`) to external ports dynamically
(such as `8081` or higher), which may require additional configuration in
TYPO3 and the reverse proxy.

When deploying TYPO3 to a container-based hosting platform, make sure that the
```

container's **internal port 80** is correctly exposed to the outside.

Most platforms handle this by routing incoming traffic to a fixed external
port (for example, port 443 for HTTPS) and forwarding it to port 80 inside the
container. You do **not** need to modify the TYPO3 image or change its
internal port.

Make sure your hosting configuration (such as ingress or service mapping)
forwards traffic to port `80` inside the container.

..  tip::

>     Most TYPO3 Docker images are configured to listen on port ``80`` inside the

Reviewer: The webserver within the image is listening on port 80 by default. Not the image

Author: I rephrased it some. But you can kind of treat containers like black boxes. True there is a webserver within but that webserver becomes part of the image

    container. This is a common convention. If your host platform or reverse
    proxy expects a different internal port, be sure to configure the Docker
    port mapping accordingly.

..  _classic-docker-reverse-proxy:

Using HTTPS and reverse proxies (SSL Offloading)
================================================

..  seealso::

    **Background reading:**

    `What is SSL Offloading? <https://certera.com/blog/know-what-is-ssl-offloading-and-how-it-works/>`_

When deploying TYPO3 behind a **reverse proxy** (such as Traefik or nginx) that
handles HTTPS termination, it is common for the proxy to **offload SSL**—
meaning the secure connection (HTTPS) is terminated at the proxy, and the
request is forwarded to the container over plain HTTP (port `80`). This

> practice is called **SSL offloading**.

Reviewer: Perfect. On point.

Author: :)

:navigation-title: Ports

..  include:: /Includes.rst.txt
..  _container-hosting-ports:

===================================================
Ports when running TYPO3 on container-based hosting
===================================================

..  include:: /Administration/Docker/Production/_Experimental.rst.txt

This section applies to running TYPO3 in managed container environments such as
Kubernetes clusters, Nexaa, AWS App Runner, or Azure App Services. These platforms
often map internal container ports (like `80`) to external ports dynamically
(such as `8081` or higher), which may require additional configuration in
TYPO3 and the reverse proxy.

When deploying TYPO3 to a container-based hosting platform, make sure that the
container's **internal port 80** is correctly exposed to the outside.

Most platforms handle this by routing incoming traffic to a fixed external
port (for example, port 443 for HTTPS) and forwarding it to port 80 inside the
container. You do **not** need to modify the TYPO3 image or change its
internal port.

Make sure your hosting configuration (such as ingress or service mapping)
forwards traffic to port `80` inside the container.

> ..  tip::
> 
>     Most TYPO3 Docker images are configured to listen on port ``80`` inside the
>     container. This is a common convention. If your host platform or reverse
>     proxy expects a different internal port, be sure to configure the Docker
>     port mapping accordingly.
> 

Author: ```suggestion
..  tip::

    TYPO3 Docker images typically expose port 80 internally. External access is managed 
    through your platform’s service or ingress configuration.

```
